pytket/tests/qasm_test.py

```diff
@@ -358,6 +350,22 @@ def test_output_error_modes() -> None:
     )
 
 
+def test_header_stops_gate_definition() -> None:
+    c = Circuit(2)
+    c.add_gate(OpType.ZZMax, [0, 1])
+    # adds a custom gate, "zzmax"
+    qasm_str_qelib1 = circuit_to_qasm_str(c, header="qelib1")
+    # adds "ZZ"
+    qasm_str_hqslib1 = circuit_to_qasm_str(c, header="hqslib1")
+    assert qasm_str_qelib1[48:76] == "gate zzmax zzmaxq0,zzmaxq1 {"
+    assert qasm_str_qelib1[335] == "}"
+    assert qasm_str_qelib1[337:353] == "zzmax q[0],q[1];"
+    assert qasm_str_hqslib1[49:62] == "ZZ q[0],q[1];"
```

Comment:
Sorry, could we leave out the indexes and just test that the substrings are there? I just think that any change that adds e.g. whitespace will break this test which will be annoying in the future.

Reply:
Ah yeahp sure, done